feat: use es2018 lib in tsconfig.json files

christianliebel opened the pull request:
Update the `lib` property in tsconfig.json files of both schematics and the DevKit repo to `es2018`.

The ECMAScript 2018 feature set was finalized in January, so it seems reasonable to upgrade the libs as well. This is specifically interesting for the new Angular libs that are stuck at ECMAScript 2015 (!).

christianliebel commented:
I signed it!

clydin commented:
Please avoid changing files within the `tests` directory.  They are generally setup to test specific scenarios.

Also, `feat` requires a scope.  The commits should ideally be split into:
* `feat(@schematics/angular): ...`
* `feat(@schematics/schematics): ...`
* `build: ...` -> for the main tsconfig

christianliebel commented:
@clydin I see. Thanks for your help. I’ll update the PR accordingly.

christianliebel committed:
build: update tsconfig.json libs to include es2018

Make the es2018 typescript lib available for the repository.

christianliebel committed:
feat(@schematics/angular): update tsconfig.json libs to include es2018

Update the lib property in tsconfig.json files for Angular libraries and workspaces to include es2018.

christianliebel committed:
feat(@schematics/schematics): update tsconfig.json libs to include es2018

Update the lib property in tsconfig.json files to include es2018.

christianliebel commented:
@clydin The tests looks better now. 👍 However, bazel-build seems to have a problem. Which TypeScript version does bazel use internally? TypeScript 2.7.2 actually has the `es2018` lib.

christianliebel commented:
I’ve created an additional PR (https://github.com/angular/devkit/pull/870) that only bumps the lib to `es2017`, which should work using the current bazel build.

filipesilva commented:
@alexeagle do you know why the bazel rules only support TS 2.6.x?

clydin commented:
The devkit is built against 2.7.  The problem appears to be that the lib type definitions are being sourced from: 
```
/home/circleci/.cache/bazel/_bazel_circleci/9ce5c2144ecf75d11717c0aa41e45a8d/execroot/angular_devkit/external/build_bazel_rules_typescript_tsc_wrapped_deps/node_modules/typescript/lib/lib.es2018.d.ts
```
instead of the typescript version used to compile:
```
/home/circleci/ng/node_modules/typescript/lib/typescript.js
```

clydin commented:
The error may stem from the following line which attempts to optimize the build time by re-using the library definitions from an existing project.

https://github.com/angular/devkit/blob/master/packages/angular_devkit/build_optimizer/BUILD#L32

alexeagle commented:
Sorry @christianliebel , this got lost when we moved the code back over to the angular/angular-cli repo. Do you have time to re-base this commit on master over there?
